cli/src/main/java/uk/ac/ebi/atlas/cli/experimentDesign/ExperimentDesignCommand.java

```diff
@@ -0,0 +1,39 @@
+package uk.ac.ebi.atlas.cli.experimentDesign;
+
+import org.springframework.stereotype.Component;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+import uk.ac.ebi.atlas.experimentimport.GxaExperimentCrud;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.logging.Logger;
+
+@Component
+@Command(
+        name = "update-experiment-design",
+        description = "Update experiment design for a set of accessions")
+public class ExperimentDesignCommand implements Callable<Integer> {
+    private static final Logger LOGGER = Logger.getLogger(ExperimentDesignCommand.class.getName());
+
+    @Option(names = {"-e", "--experiment"}, split = ",", description = "one or more experiment accessions", required = true)
+    private List<String> experimentAccessions;
+
+    private final GxaExperimentCrud experimentCrud;
+
+    public ExperimentDesignCommand(GxaExperimentCrud gxaExperimentCrud) {
+        this.experimentCrud = gxaExperimentCrud;
+    }
+
+    @Override
+    public Integer call() {
+
+        LOGGER.info("Starting update experiment designs for accessions.");
+        for(String accession : experimentAccessions) {
+            experimentCrud.updateExperimentDesign(accession);
+        }
```

Comment:
It seems to be more code and longer to read, but happy to go with that for coherence of what you do elsewhere. What limits the number of threads in parallel? Would this work:

```
-Djava.util.concurrent.ForkJoinPool.common.parallelism=16
```
? Or there is a newer way of doing that?

I suspect that it will take by default something related to the number of cores in the machine, which on the cluster can be very high and go beyond the LSF allocation I suspect. I worry since we will be sending this with a lot experiments.

Reply:
Yes, that would be the way to go. If not specified, as you suspect, it will create as many threads as CPU cores are available to the JVM.